### PR TITLE
Added GetSupportedOptimizationTypes API and changed the optimizationtype to an enum

### PR DESCRIPTION
**Pull request by prakharsharma10**
In this PR, we have updated the swagger for the API version 2016-10-02. 

The changes include the following:
1. Added GetSupportedOptimizationTypes API
2. Changed the optimization type from a string to an enum

For these changes in the swagger, we do not plan to generate the SDK and write test cases as we'll soon generate the SDK using the 2017-04-02 version of swagger.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by alvadb — CHANGES_REQUESTED** [review]
Please fix linter issues:  https://travis-ci.org/Azure/azure-rest-api-specs/jobs/262362880

File: specification/cdn/resource-manager/Microsoft.Cdn/2016-10-02/cdn.json
Comment (ravbhatnagar) [inline specification/cdn/resource-manager/Microsoft.Cdn/2016-10-02/cdn.json]: @prakharsharma10 - Why isnt optimizationTypes a property on profiles resource type? So GET on /profiles/{profileName} contains this property in the response payload. If you dont want it to be included by defauls in the payload, You can consider exposing it using a query parameter $expand "supportedOptimizationTypes".
Do you want to ACL it explicitly through the POST action and hence you are doing it this way? Please help me understand the scenario.

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]

File: specification/cdn/resource-manager/Microsoft.Cdn/2016-10-02/cdn.json
Comment (prakharsharma10) [inline specification/cdn/resource-manager/Microsoft.Cdn/2016-10-02/cdn.json]: Each profile has a SKU associated with it like Standard_Akamai, Premium_Verizon etc and we need to update the supported optimization types for a SKU from time to time based on the features provided by the providers. Therefore, if we have it as a property of a profile, we'll need to update all the profiles in the datastore every time we update the supported optimization types for a SKU. So, to keep it dynamic we have exposed the method 'GetSupportedOptimizationTypes' for profiles.

Comment (ravbhatnagar) [inline specification/cdn/resource-manager/Microsoft.Cdn/2016-10-02/cdn.json]: @prakharsharma10 - Ok, just one thing to note here is that this will be ACL'd via a POST action. So, someone with a read access to a profile will not be able to get the supported optimization types for that profile. He will need to have access to this POST action.

Comment (prakharsharma10) [inline specification/cdn/resource-manager/Microsoft.Cdn/2016-10-02/cdn.json]: @ravbhatnagar That is true. We should not have any issues with access control because all the CDN users can make get/put/patch calls.
I believe, we're now waiting for the ARM team to sign-off.

**Comment by ravbhatnagar** [comment]
@prakharsharma10 - Sounds good. @alvadb - ARM signs off.

**Review by alvadb — APPROVED** [review]

**Comment by alvadb** [comment]
@ravbhatnagar please approve your requested changes.

**Review by ravbhatnagar — APPROVED** [review]

**Comment by prakharsharma10** [comment]
@alvadb Can you please merge the PR? Thanks :)

**Comment by salameer** [comment]
@alvadb please merge if you're good on your side

**Comment by prakharsharma10** [comment]
@alvadb @fearthecowboy @salameer gentle ping!

**Pull request merged**
